model/project.go

```diff
@@ -1360,6 +1360,16 @@ func FindLatestVersionWithValidProject(projectId string) (*Version, *Project, *P
 		}
 
 		env := evergreen.GetEnvironment()
+		if preGeneration {
+			preGeneratedId := fmt.Sprintf("%s_%s", "pre_generation", lastGoodVersion.Id)
+			pp, err = ParserProjectFindOneByID(ctx, env.Settings(), lastGoodVersion.ProjectStorageMethod, fmt.Sprintf("%s_%s", "pre_generation", lastGoodVersion.Id))
```

Comment:
It seems a little redundant to load the parser project twice here, once just to confirm that it exists, and then once for real on L1373. Could we consider doing something a little smarter, like once we know the pre-generated parser project exists and is non-nil, just call `TranslateProject` on it? And only if it was unable to find one, then fall back to finding + translating the post-generated parser project?

Reply:
Refactored this logic to add a bool flag to `FindAndTranslateProjectForVersion` which will consolidate this into one DB call in the happy path scenario.